### ISSUE TITLE
Docs failing to load due to markdown parser breaking changes

### ISSUE THREAD

**Issue by jakeswenson** [rust-lang/rustfmt#6197]
The rustfmt docs site is failing to load for me. 

I've tracked it down to what looks like a change in `marked`. 

Specifically the heading function is now handed a single object argument instead of what the current code is expecting.

In marked:
```javascript
case 'heading': {
  out += this.renderer.heading(token);
  continue;
}
```
https://github.com/markedjs/marked/blob/70bb55e0af5128a657a14b8b25d7d406661e6936/src/Parser.ts#L69-L71

Marked made an API contract breaking change for this: https://github.com/markedjs/marked/pull/3291

In rustfmt page:
https://github.com/rust-lang/rustfmt/blob/c97996fab61fc09432abcf64dbe3ce2d84f4d679/docs/index.html#L172-L178

`htmlToId` is now failing because text is an object, and doesn't have a `.trim()` method
https://github.com/rust-lang/rustfmt/blob/c97996fab61fc09432abcf64dbe3ce2d84f4d679/docs/index.html#L304-L308

**Comment by jakeswenson** [comment]
Quickest fix would be to pin to a proper major version for `marked`
https://github.com/rust-lang/rustfmt/blob/c97996fab61fc09432abcf64dbe3ce2d84f4d679/docs/index.html#L8

**Comment by jakeswenson** [comment]
Fixed in #6194
